zerver/tests/test_embedded_bot_system.py

```diff
@@ -57,6 +60,42 @@ def test_stream_message_not_to_embedded_bot(self) -> None:
         last_message = self.get_last_message()
         self.assertEqual(last_message.content, "foo")
 
+    def test_embedded_bot_react_to_stream_message(self) -> None:
+        assert self.bot_profile is not None
+        self.send_stream_message(
+            self.user_profile,
+            "Denmark",
+            content=f"@**{self.bot_profile.full_name}** foo",
+            topic_name="bar",
+        )
+        user_message = self.get_second_to_last_message()
+        self.assertTrue(
+            Reaction.objects.filter(
+                message=user_message,
+            ).exists()
+        )
+
+    def test_embedded_bot_react_to_pm(self) -> None:
+        assert self.bot_profile is not None
+        self.send_personal_message(self.user_profile, self.bot_profile, content="help")
+        user_message = self.get_second_to_last_message()
+        self.assertTrue(
+            Reaction.objects.filter(
+                message=user_message,
+            ).exists()
+        )
```

Comment:
Where is the test setup logic that configures the bot in question to actually call `react()`?

Reply:
The bot is a helloworld bot, which replies to the message and adds a reaction to it. I will note that in the comment.